### ISSUE TITLE
[bug]: Cannot login on FreePBX 17 Administration

### ISSUE THREAD

**Issue by AppleCarmine** [FreePBX/issue-tracker#402]
### FreePBX Version

FreePBX 17

### Issue Description

I installed FreePBX 17 on Debian 12 in a clean VPS. The installation went well, but when I try to log in to “FreePBX Administration,” I get this in Chrome’s console and I can’t log in:
![Screenshot 2024-08-29 alle 19 00 54](https://github.com/user-attachments/assets/59c02c27-54e7-47c1-a1e3-ff5b35103b66)

### Operating Environment

FreePBX 17.0.19.7 
| accountcodepreserve | 17.0.0.1  | Enabled | GPLv2       | Sangoma   |
| adv_recovery        | 17.0.4    | Enabled | Commercial  | Sangoma   |
| allowlist           | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| amd                 | 17.0.1    | Enabled | GPLv3+      | Sangoma   |
| announcement        | 17.0.2.1  | Enabled | GPLv3+      | Sangoma   |
| api                 | 17.0.1.6  | Enabled | AGPLv3+     | Sangoma   |
| areminder           | 17.0.3.10 | Enabled | Commercial  | Sangoma   |
| arimanager          | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| asterisk-cli        | 17.0.2    | Enabled | GPLv3+      | Sangoma   |
| asteriskinfo        | 17.0.2    | Enabled | GPLv3+      | Sangoma   |
| backup              | 17.0.5.60 | Enabled | GPLv3+      | Sangoma   |
| blacklist           | 17.0.1.2  | Enabled | GPLv3+      | Sangoma   |
| broadcast           | 17.0.1.6  | Enabled | Commercial  | Sangoma   |
| builtin             |           | Enabled |             | Unsigned  |
| bulkhandler         | 17.0.5    | Enabled | GPLv3+      | Sangoma   |
| calendar            | 17.0.4.18 | Enabled | GPLv3+      | Sangoma   |
| callaccounting      | 17.0.5    | Enabled | Commercial+ | Sangoma   |
| callback            | 17.0.2.1  | Enabled | GPLv3+      | Sangoma   |
| callerid            | 17.0.1    | Enabled | Commercial  | Sangoma   |
| callforward         | 17.0.1.6  | Enabled | AGPLv3+     | Sangoma   |
| calllimit           | 17.0.1.2  | Enabled | Commercial  | Sangoma   |
| callrecording       | 17.0.3.7  | Enabled | AGPLv3+     | Sangoma   |
| callwaiting         | 17.0.3.4  | Enabled | GPLv3+      | Sangoma   |
| cdr                 | 17.0.4.20 | Enabled | GPLv3+      | Sangoma   |
| cdrpro              | 17.0.3.19 | Enabled | Commercial  | Sangoma   |
| cel                 | 17.0.2.8  | Enabled | GPLv3+      | Sangoma   |
| certman             | 17.0.3.13 | Enabled | AGPLv3+     | Sangoma   |
| cidlookup           | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| conferences         | 17.0.3.2  | Enabled | GPLv3+      | Sangoma   |
| conferencespro      | 17.0.1.7  | Enabled | Commercial  | Sangoma   |
| configedit          | 17.0.1.2  | Enabled | AGPLv3+     | Sangoma   |
| contactmanager      | 17.0.5.11 | Enabled | GPLv3+      | Sangoma   |
| core                | 17.0.14   | Enabled | GPLv3+      | Sangoma   |
| cos                 | 17.0.1.1  | Enabled | Commercial  | Sangoma   |
| customappsreg       | 17.0.1    | Enabled | GPLv3+      | Sangoma   |
| customcontexts      | 17.0.1.3  | Enabled | GPLv2+      | Sangoma   |
| dashboard           | 17.0.4.4  | Enabled | AGPLv3+     | Sangoma   |
| daynight            | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| dictate             | 17.0.1.2  | Enabled | GPLv3+      | Sangoma   |
| directory           | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| disa                | 17.0.4    | Enabled | AGPLv3+     | Sangoma   |
| donotdisturb        | 17.0.2.3  | Enabled | GPLv3+      | Sangoma   |
| dynroute            | 17.0.3.2  | Enabled | GPLv3+      | Sangoma   |
| endpoint            | 17.0.1.81 | Enabled | Commercial  | Sangoma   |
| extensionroutes     | 17.0.1    | Enabled | Commercial  | Sangoma   |
| extensionsettings   | 17.0.1    | Enabled | GPLv3+      | Sangoma   |
| fax                 | 17.0.3.4  | Enabled | GPLv3+      | Sangoma   |
| faxpro              | 17.0.1.18 | Enabled | Commercial  | Sangoma   |
| featurecodeadmin    | 17.0.2    | Enabled | GPLv3+      | Sangoma   |
| filestore           | 17.0.2.23 | Enabled | AGPLv3      | Sangoma   |
| findmefollow        | 17.0.4.9  | Enabled | GPLv3+      | Sangoma   |
| firewall            | 17.0.1.26 | Enabled | AGPLv3+     | Sangoma   |
| framework           | 17.0.19.7 | Enabled | GPLv2+      | Sangoma   |
| hotelwakeup         | 17.0.1.6  | Enabled | GPLv2       | Sangoma   |
| iaxsettings         | 17.0.1    | Enabled | AGPLv3      | Sangoma   |
| infoservices        | 17.0.1.1  | Enabled | GPLv2+      | Sangoma   |
| ivr                 | 17.0.6    | Enabled | GPLv3+      | Sangoma   |
| languages           | 17.0.1    | Enabled | GPLv3+      | Sangoma   |
| logfiles            | 17.0.3.3  | Enabled | GPLv3+      | Sangoma   |
| manager             | 17.0.6    | Enabled | GPLv2+      | Sangoma   |
| miscapps            | 17.0.3    | Enabled | GPLv3+      | Sangoma   |
| miscdests           | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| missedcall          | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| music               | 17.0.5    | Enabled | GPLv3+      | Sangoma   |
| outcnam             | 17.0.1    | Enabled | GPLv3+      | Sangoma   |
| outroutemsg         | 17.0.1    | Enabled | GPLv3+      | Sangoma   |
| paging              | 17.0.3    | Enabled | GPLv3+      | Sangoma   |
| pagingpro           | 17.0.1.6  | Enabled | Commercial  | Sangoma   |
| parking             | 17.0.2.5  | Enabled | GPLv3+      | Sangoma   |
| parkpro             | 17.0.1.4  | Enabled | Commercial  | Sangoma   |
| pbxmfa              | 17.0.2.1  | Enabled | Commercial+ | Sangoma   |
| phpinfo             | 17.0.1    | Enabled | GPLv2+      | Sangoma   |
| pinsets             | 17.0.3.2  | Enabled | GPLv3+      | Sangoma   |
| pinsetspro          | 17.0.2    | Enabled | Commercial  | Sangoma   |
| pm2                 | 17.0.3.2  | Enabled | AGPLv3+     | Sangoma   |
| pms                 | 17.0.2.18 | Enabled | Commercial  | Sangoma   |
| presencestate       | 17.0.2.4  | Enabled | GPLv3+      | Sangoma   |
| printextensions     | 17.0.1.2  | Enabled | GPLv3+      | Sangoma   |
| queueprio           | 17.0.1.4  | Enabled | GPLv3+      | Sangoma   |
| queues              | 17.0.1.11 | Enabled | GPLv2+      | Sangoma   |
| queuestats          | 17.0.1.7  | Enabled | Commercial  | Sangoma   |
| qxact_reports       | 17.0.3    | Enabled | Commercial  | Sangoma   |
| recording_report    | 17.0.3.8  | Enabled | Commercial  | Sangoma   |
| recordings          | 17.0.2.2  | Enabled | GPLv3+      | Sangoma   |
| restapps            | 17.0.1.21 | Enabled | Commercial  | Sangoma   |
| ringgroups          | 17.0.2.4  | Enabled | GPLv3+      | Sangoma   |
| sangomaconnect      | 17.0.1.35 | Enabled | Commercial  | Sangoma   |
| sangomacrm          | 17.0.1.16 | Enabled | Commercial  | Sangoma   |
| sangomartapi        | 17.0.1.28 | Enabled | Commercial  | Sangoma   |
| setcid              | 17.0.1.2  | Enabled | GPLv3+      | Sangoma   |
| sipsettings         | 17.0.6.9  | Enabled | AGPLv3+     | Sangoma   |
| sipstation          | 17.0.3.3  | Enabled | Commercial  | Sangoma   |
| sms                 | 17.0.1.15 | Enabled | Commercial  | Sangoma   |
| smsplus             | 17.0.3    | Enabled | Commercial  | Sangoma   |
| soundlang           | 17.0.4.1  | Enabled | GPLv3+      | Sangoma   |
| superfecta          | 17.0.2.1  | Enabled | GPLv2+      | Sangoma   |
| sysadmin            | 17.0.1.95 | Enabled | Commercial  | Sangoma   |
| timeconditions      | 17.0.1.18 | Enabled | GPLv3+      | Sangoma   |
| tts                 | 17.0.1.1  | Enabled | GPLv3+      | Sangoma   |
| ttsengines          | 17.0.1    | Enabled | AGPLv3      | Sangoma   |
| ucp                 | 17.0.4.21 | Enabled | AGPLv3+     | Sangoma   |
| userman             | 17.0.6.27 | Enabled | AGPLv3+     | Sangoma   |
| vmblast             | 17.0.2    | Enabled | GPLv3+      | Sangoma   |
| vmnotify            | 17.0.1.7  | Enabled | Commercial  | Sangoma   |
| voicemail           | 17.0.5.22 | Enabled | GPLv3+      | Sangoma   |
| voicemail_report    | 17.0.1.1  | Enabled | Commercial  | Sangoma   |
| voipinnovations     | 17.0.1.4  | Enabled | Commercial  | Sangoma   |
| vqplus              | 17.0.1.17 | Enabled | Commercial  | Sangoma   |
| weakpasswords       | 17.0.1    | Enabled | GPLv3+      | Sangoma   |
| webcallback         | 17.0.4    | Enabled | Commercial  | Sangoma   |
| webrtc              | 17.0.2.2  | Enabled | GPLv3+      | Sangoma   |
+---------------------+-----------+---------+-------------+-----------+

### Relevant log output

```shell
Code of the adminPwdExpReminder.js:8 (kgupta asked me to put this on github)

function checkPasswordReminder(thisPointerFromLogin) {

    return new Promise(function (resolve, reject) {

        window.loginFormPointer = thisPointerFromLogin;
        window.username = $('input[name="username"]', thisPointerFromLogin).val();
        window.password = $('input[name="password"]', thisPointerFromLogin).val();
        window.password = encodeURIComponent(window.btoa(window.password))

        $.post("/admin/ajax.php?module=userman&command=checkPasswordReminder",
            {
                username: window.username,
                password: window.password,
                loginpanel: 'admin'
            }).done(function (response) {

                if (response.isSessionAlreadyUnlocked) {

                    window.location.reload();

                } else if (response.loginfailed) {

                    fpbxToast(_(response.message), _('Error'), "error");

                } else if (response.mustresetpassword) {

                    alert(_(response.message));

                    if (response.usertype && response.usertype == 'ucp') {
                        setTimeout(() => {
                            window.location.href = response.resetlink
                        }, 300);
                    } else {

                        // Displaying Reset form
                        $('.ui-dialog-content').html(`
                            <div id="reset_form">
                                <h5> ${_('Your password has been expired. Please reset it to continue.')}</h5>
                                <div class="form-group">
                                    <input type="password" id="newPassword" name="newPassword" class="form-control" value="" placeholder="Password" autocomplete="off">
                                </div>
                                <div class="form-group">
                                    <input type="password" id="confirmPaswword" name="confirmPaswword" class="form-control" value="" placeholder="Confirm Paswword" autocomplete="off">
                                </div>
                            </div>
                        `);

                        $('.ui-dialog-titlebar').text('Reset Password');
                        $('div.ui-dialog-buttonpane.ui-widget-content.ui-helper-clearfix > div > button:nth-child(1)').addClass('resetPasswordButton');

                        window.resetToken = response.resetPasswordToken;
                    }

                } else {

                    if (!response.status) {
                        alert(_(response.message));
                    }

                    resolve(true);
                }
            }).fail(function (xhr, status, error) {
                fpbxToast(_(error), _('Error'), "error");
                reject(true);
            });
    })
}

$('#reset_form input[name="newPassword"]').on('keyup', function () {
    if ($('#reset_form input[name="newPassword"]').val() != $('#reset_form input[name="confirmPaswword"]').val()) {
        fpbxToast(_('Confirm password does not match'), _("Error"), 'error');
    }
});

function resetAdminPassswordWithToken(thisPointerFromLogin) {
    window.newPassword = $('#reset_form input[name="newPassword"]', thisPointerFromLogin).val();
    window.confirmPassword = $('#reset_form input[name="confirmPaswword"]', thisPointerFromLogin).val();

    if (window.newPassword == '' || window.confirmPassword == '') {
        fpbxToast(_('Invalid Password'), _('Error'), 'error');
        return false;
    }
    if ($('#newPassword').val() != $('#confirmPaswword').val()) {
        fpbxToast(_('Confirm password does not match'), _('Error'), 'error');
        return false;
    }

    $.post("/admin/ajax.php?module=userman&command=resetAdminPasswordWithToken",
        {
            newpassword: window.newPassword,
            token: window.resetToken,
            confirmpassword: window.confirmPassword,
        }).done(function (response) {
            if (response.status) {
                fpbxToast(_(response.message));
            } else {
                fpbxToast(_(response.message), _('Error'), "error");
            }
            setTimeout(() => {
                window.location.reload();
            }, 300);
        }).fail(function (xhr, status, error) {
            fpbxToast(_(error), _('Error'), "error");
        });

};
```

**Comment by nkovats** [comment]
**FreePBX Version**

FreePBX 17

**Issue Description:**

Similar issue. Cannot log onto FreePBX WebUI. 

Installed FreePBX 17 on Debian 12.6.00-23-arm64 on ASRock SBC-210 server.  Debian 12 installed cleanly and it's OS UI is working as is SSH server. FreePBX Installation log (attached) does not appear to have errors but appears to be incomplete. I tried 4x different browsers which keep hitting an Apache splash page (attached). Screenshot taken with Chrome Version 128.0.6613.113.

![08-29-24 Apache2 Debian Default Page](https://github.com/user-attachments/assets/f500e84a-460a-44f1-b134-c63da4a20fa7)
[08-29-24 freepbx install log.txt](https://github.com/user-attachments/files/16803744/08-29-24.freepbx.install.log.txt)

**Comment by kapilgupta01** [comment]
Hi @nkovats your error is different, you are getting default apache page which installation script is removing ref -https://github.com/FreePBX/sng_freepbx_debian_install/blob/master/sng_freepbx_debian_install.sh#L1151  so please check if the installation is successfully done or not. 

You could manually remove the file to fix your issue.

**Comment by kapilgupta01** [comment]
Hi @AppleCarmine Is it possible for you to give access to your system to us so we can debug your issue?
If yes please share the credentials to kgupta@sangoma.com

Thanks

**Comment by kapilgupta01** [comment]
Hi @AppleCarmine Waiting for your access? because this issue is not reproducible for us? 
thanks

**Comment by EmnaX** [comment]
I was running into a similar issue... couldn't login. I took a screenshot 
![Screenshot 2024-09-11 at 2 28 18 PM](https://github.com/user-attachments/assets/f7589acf-39c6-4b92-bb18-3b6d704967c3)

From the screenshot, the error was "Uncaught (in promise) DOMException: String contains an invalid character".

I started playing around with my password and found that there was one character in there that was causing the issue. My original password which caused the error was: 62q4Q)S1-6_{dUf3)nU’7AaO=

The character which was causing the error was: ’

After using `fwconsole unlock` to get me access to the WebUI, I was able to change my admin password to something else and then logging in worked as expected.
